### ISSUE TITLE
Registry hive parsing issue

### ISSUE THREAD

**Issue by thomaslaurenson** [thomaslaurenson/CellXML-Registry#1]
Encountered a bug when parsing and recovering deleted entries for a number of Window Registry hive files using any software based on the Registry parser project (https://github.com/EricZimmerman/Registry). Basically, when processing the hive file the application hangs, but only when specifying to recovered deleted entries. The following documents the bug with different applications and settings:
## Causes application to hang

1) CellXML-Registry-1.0.0.exe -r -f {hive}
2) RECmd.exe --Hive {hive} --Recover
3) ExampleApp.exe -f {hive} -r
4) Loading {hive} into Registry Explorer (version 0.7.1.0)
## DOES NOT cause application to hang

1) CellXML-Registry-1.0.0.exe  -f {hive}
2) RECmd.exe --Hive {hive}
3) ExampleApp.exe -f {hive}

Problem:
Basically, it seems that whenever one of the 'problematic' Registry hive files is processed, the processing stage only hangs when attempting to recover Registry entries.

I have uploaded two problematic hive files: 
https://github.com/thomaslaurenson/CellXML-Registry/tree/master/problem_hives

Solution:
Contact Registry parser project author Eric Zimmerman.
Do a detailed review of debug output when parsing, add in extra debug messages if needed.

Additional Note: The last log file entry is:
BuildDeletedRegistryKeys INFO Associating deleted keys and values...
